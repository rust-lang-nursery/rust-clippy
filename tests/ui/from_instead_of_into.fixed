// run-rustfix
#![allow(unused_imports)]
#![warn(clippy::from_instead_of_into)]
use std::convert::TryFrom;
use std::convert::TryInto;

fn foo<T>(a: T)
where
    T: Into<u32>,
{
}

fn foo1<T>(a: T)
where T: Into<u32>, u32: Copy + Clone,
{
}

fn bar<T>(a: T)
where
    T: TryInto<u32>,
{
}

fn bar1<T>(a: T)
where
    T: TryInto<u32>, u32: Copy + Clone,
{
}

fn bar2<T>(a: T)
where
    T: TryInto<u32>, u32: Copy + Clone,
{
}

fn main() {}
