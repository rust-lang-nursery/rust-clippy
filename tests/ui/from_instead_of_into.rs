// run-rustfix
#![allow(unused_imports)]
#![allow(unused_variables)]
#![allow(dead_code)]
#![warn(clippy::from_instead_of_into)]
use std::convert::TryFrom;
use std::convert::TryInto;

fn foo<T>(a: T)
where
    u32: From<T>,
{
}

fn foo1<T>(a: T)
where
    u32: Copy + Clone + From<T>,
{
}

fn bar<T>(a: T)
where
    u32: TryFrom<T>,
{
}

fn bar1<T>(a: T)
where
    u32: Copy + TryFrom<T> + Clone,
{
}

fn bar2<T>(a: T)
where
    u32: TryFrom<T> + Copy + Clone,
{
}

fn main() {}
