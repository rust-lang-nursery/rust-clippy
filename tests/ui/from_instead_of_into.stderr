error: Into trait is preferable than From as a generic bound
  --> $DIR/from_instead_of_into.rs:11:5
   |
LL |     u32: From<T>,
   |     ^^^^^^^^^^^^ help: try: `T: Into<u32>`
   |
   = note: `-D clippy::from-instead-of-into` implied by `-D warnings`

error: Into trait is preferable than From as a generic bound
  --> $DIR/from_instead_of_into.rs:17:5
   |
LL |     u32: Copy + Clone + From<T>,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `T: Into<u32>, u32: Copy + Clone`

error: TryInto trait is preferable than TryFrom as a generic bound
  --> $DIR/from_instead_of_into.rs:23:5
   |
LL |     u32: TryFrom<T>,
   |     ^^^^^^^^^^^^^^^ help: try: `T: TryInto<u32>`

error: TryInto trait is preferable than TryFrom as a generic bound
  --> $DIR/from_instead_of_into.rs:29:5
   |
LL |     u32: Copy + TryFrom<T> + Clone,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `T: TryInto<u32>, u32: Copy + Clone`

error: TryInto trait is preferable than TryFrom as a generic bound
  --> $DIR/from_instead_of_into.rs:35:5
   |
LL |     u32: TryFrom<T> + Copy + Clone,
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: try: `T: TryInto<u32>, u32: Copy + Clone`

error: aborting due to 5 previous errors

